autograder/plagiarism_detection/plagiarism_detection.py

```diff
@@ -1,50 +1,96 @@
         return "No files found"
-    prev = files[0].name.split('.')[1]
+    lexerClass, ignoreList, numTokens = checkFileTypes(files)
+    similarityScores = [[0 for x in range(numFiles)] for y in range(numFiles)]
+    # get token stream for each file and total frequency for each token type
+    tokenStreams, tokenFrequencies = parseFiles(
+        files, lexerClass, ignoreList, numTokens
+    )
+    similarityMatrix = buildSimilarityMatrix(tokenFrequencies)
+    print(similarityMatrix)
     for i in range(numFiles):
-        fileType = files[i].name.split('.')[1]
+        for j in range(i + 1, numFiles):
+            similarity = getSimilarity(tokenStreams[i], tokenStreams[j])
+            similarityScores[i][j] = similarity
+            similarityScores[j][i] = similarity
+    return similarityScores
+
+
+# lexerClass is the lexer to be used, ignoreList is a list containing the
+# token types to remove when parsing, numTokens is the number of token types in the language
+def checkFileTypes(files):
+    prev = files[0].name.split(".")[1]
+    for i in range(len(files)):
+        fileType = files[i].name.split(".")[1]
         if fileType != prev:
             return "File types don't match"
         prev = fileType
         if fileType == "java":
             lexerClass = JavaLexer
             # ignore comments
-            ignoreList = [107]
+            ignoreList = [-1, 106, 107]
+            numTokens = 105
         elif fileType == "py":
             lexerClass = Python3Lexer
             # ignore comments and new lines
             ignoreList = [-1, 39]
-        else:
-            return "This file type is not supported"
-    similarityScores = [[0 for x in range(numFiles)] for y in range(numFiles)]
+            numTokens = 97
+    return lexerClass, ignoreList, numTokens
+
+
+def parseFiles(files, lexerClass, ignoreList, numTokens):
     tokenStreams = []
-    for i in range(numFiles):
+    for i in range(len(files)):
         stream = InputStream(files[i].read())
         files[i].close()
         lexer = lexerClass(stream)
         tokens = CommonTokenStream(lexer)
         tokens.fetch(500)
         array = []
+        freq = [0 for x in range(numTokens + 1)]
+        totalTokens = 0
         for token in tokens.tokens:
             # remove comments
-            print(token.type)
             if token.type not in ignoreList:
                 array += [str(token.type)]
+                freq[token.type] += 1
+                totalTokens += 1
         tokenStreams += [array]
-    for i in range(numFiles):
-        for j in range(i + 1, numFiles):
-            similarity = getSimilarity(tokenStreams[i], tokenStreams[j])
-            similarityScores[i][j] = similarity
-            similarityScores[j][i] = similarity
-    return similarityScores
+        for i in range(len(freq)):
+            freq[i] = max(freq[i] / totalTokens, 0.0000000001)
+    return tokenStreams, freq
+
+
+def buildSimilarityMatrix(freq):
+    alpha = 0.65
```

Comment:
Either improve naming or use docstrings/comments to describe what this function does and why we need alpha and beta.